### DISCUSSION TITLE
Not lock runtime python

### DISCUSSION THREAD

**Discussion by leandrogmuller** [sqlalchemy/sqlalchemy#6312] [Usage Questions]
I need to execute merge, but not lock object during the process.

engine = create_engine("postgresql://user:pass@localhost:5432/datadev",poolclass=NullPool)

for i in range(3): 
    session_factory = sessionmaker(bind=engine)
    session = session_factory()
    lang03 = session.query(Lang03).filter(Lang03.pkge == 'kernel').first()
    lang03.text = 'aa'
    session.merge(lang03) #second attempt lock the code running. I dont't want lock the runtime.

**Comment by CaselIT** [comment]
Hi,

I'm not sure why you would want to merge an object that's already in the session. See the documentation for details https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=merge#sqlalchemy.orm.Session.expunge

**Reply by zzzeek** [reply]
well also it shouldn't "lock" anything, what kind of "lock" are we referring towards?   a database lock or some kind of threading mutex is invovled somewhere?

**Reply by leandrogmuller** [reply]
Hi.
I saw lock on database. The session is locking database record before commit.

**Reply by zzzeek** [reply]
SQLAlchemy just emits the SQL you give it.  the psycopg2 driver will emit BEGIN TRANSACTION ahead of that. turn on Postgresql [SQL logging](https://www.postgresql.org/docs/9.4/runtime-config-logging.html) and you'll be able to see all commands being received by the database.
